metaswitch/common/utils.py

```diff
@@ -399,3 +400,22 @@ def sigusr1_handler(sig, stack):
         stack_dump = "Caught SIGUSR1\n" + "".join(traceback.format_stack(stack))
         write_core_file(process_name, stack_dump)
     signal.signal(signal.SIGUSR1, sigusr1_handler)
+
+def write_pid_file(filename):
```

Comment:
I'll rename it to `lock_and_write_pidfile` - `acquire_pid_file_lock` might look as though the caller still has to write the pid.